fawltydeps/settings.py

```diff
@@ -238,12 +236,43 @@ def print_toml_config(settings: Settings, out: TextIO = sys.stdout) -> None:
         logger.critical(f"Sanity check failed: {settings!r} is missing a field!")
         raise
 
-    lines = [
-        "# Copy this TOML section into your pyproject.toml to configure FawltyDeps",
-        "# (default values are commented)",
-        "[tool.fawltydeps]",
-    ] + [
-        f"{'# ' if has_default_value[name] else ''}{name} = {value!r}"
-        for name, value in simple_settings.items()
-    ]
+    dictionary_options = ["custom_mapping"]
+
+    def _option_to_toml(name, value) -> str:  # type: ignore
+        """Serialize options to toml configuration entries
+
+        Options that are of dictionary type must be given a section entry.
+        Assumption: dictionaries options are not nested.
+        """
+        comment_char = "# " if has_default_value[name] else ""
+        if name in dictionary_options:
+            toml_option = f"{comment_char}[tool.fawltydeps.{name}]"
+            if value is not None:
+                toml_option += "".join(
+                    [f"\n{comment_char}{k} = {v!r}" for k, v in value.items()]
+                )
+        else:
+            toml_option = f"{comment_char}{name} = {value!r}"
+        return toml_option
+
+    lines = (
+        [
+            "# Copy this TOML section into your pyproject.toml to configure FawltyDeps",
+            "# (default values are commented)",
+            "[tool.fawltydeps]",
+        ]
+        + [
+            _option_to_toml(name, value)
+            for name, value in simple_settings.items()
+            # First handle non-dictionary options,
+            # as rest will be treated as members of a `name` dictionary
```

Comment:
Last part of this comment does not make sense to me:
```suggestion
            # First handle non-dictionary options, as we don't want them to end
            # up inside the config section resulting from a dictionary option.
```